lib/executor.js

```diff
@@ -1,23 +1,41 @@
-var EventEmitter = require('./emitter').EventEmitter;
-var _ = require('lodash');
-var util = require('util');
-var selfAddressed = require('self-addressed');
+const _ = require('lodash');
+const util = require('util');
+const selfAddressed = require('self-addressed');
+const EventEmitter = require('./emitter').EventEmitter;
+const log = require('./logging');
+
+function LoggerProxy(logger) {
+    for (let type of ['trace', 'debug', 'info', 'warn', 'error', 'fatal']) {
+        this[type] = function log() {
+            const args = Array.prototype.slice.call(arguments);
+
+            if (args.length && typeof args[0] !== 'string') {
+                args[0] = String(args[0]);
```

Comment:
may be better to json.stringify or some other serialization?
because in case of object as argument you'll got `[object]` instead of some meaningful data

Reply:
One more idea: may be it would be more convenient to use Proxy? (i have no experience with it, but this looks like good usage)